x-pack/platform/plugins/shared/streams/server/routes/content/route.ts

```diff
@@ -110,29 +143,27 @@ const importContentRoute = createServerRoute({
 
     await streamsClient.ensureStream(params.path.name);
 
-    const body: ContentPack = await new Promise((resolve, reject) => {
-      let data = '';
-      params.body.content.on('data', (chunk) => (data += chunk));
-      params.body.content.on('end', () => {
-        try {
-          const parsed = JSON.parse(data);
-          resolve(contentPackSchema.parse(parsed));
-        } catch (err) {
-          reject(new StatusError('Invalid content pack format', 400));
-        }
-      });
-      params.body.content.on('error', (error) => reject(error));
+    const contentPack = await parseArchive(params.body.content);
+
+    const importer = (await context.core).savedObjects.getImporter(soClient);
+
+    // this is required to support integration package's dashboards. since they
+    // reference pre-existing metrics-* and logs-* data views we they need to
+    // install them before import
+    await installManagedIndexPattern({
```

Comment:
I wonder whether we can resolve this in a more elegant way - this approach will always add these managed data views in there even if they are not relevant. Maybe we can check whether they have these dangling references and only install them in this case?

I see how we don't get around some BWC stuff with this kind of work, but let's try to keep the happy path clean of it where possible.

Reply:
I didn't mind because that's a fairly low performance and storage cost even if they are not required, but I can add a check

Reply:
Not worried about performance so much, it's more about having weird and unexpected saved objects

Reply:
Updated in https://github.com/elastic/kibana/pull/217288/commits/ac1e315d7705179e1e6b0eea2b74f34fbc52025a